backend/pkg/rest/apiinventory_upload_spec.go

```diff
@@ -37,27 +36,32 @@ import (
 
 func (s *Server) PutAPIInventoryAPIIDSpecsProvidedSpec(params operations.PutAPIInventoryAPIIDSpecsProvidedSpecParams) middleware.Responder {
 	apiInfo := &database.APIInfo{}
-	var jsonSpec []byte
-	var err error
 
 	log.Debugf("Got PutAPIInventoryAPIIDSpecsProvidedSpecParams: %+v", params)
 
-	jsonSpec = []byte(params.Body.RawSpec)
+	// Convert YAML to JSON. Since JSON is a subset of YAML, passing JSON through
+	// this method should be a no-op.
+	jsonSpecBytes, err := yaml.YAMLToJSON([]byte(params.Body.RawSpec))
+	if err != nil {
+		log.Errorf("Failed to convert yaml spec to json: %s. %v", params.Body.RawSpec, err)
+		return operations.NewPutAPIInventoryAPIIDSpecsProvidedSpecDefault(http.StatusInternalServerError)
+	}
 
-	// if spec is a yaml spec, convert it to json format for saving in db
-	if isYamlSpec([]byte(params.Body.RawSpec)) {
-		jsonSpec, err = yaml.YAMLToJSON([]byte(params.Body.RawSpec))
-		if err != nil {
-			// The spec was already validated as a valid yaml, so error here is an internal error, not a validation error
-			log.Errorf("Failed to convert yaml spec to json: %s. %v", params.Body.RawSpec, err)
-			return operations.NewPutAPIInventoryAPIIDSpecsProvidedSpecDefault(http.StatusInternalServerError)
-		}
+	// Creates a new analyzed spec document for the provided spec
+	analyzed, err := loads.Analyzed(jsonSpecBytes, "")
+	if err != nil {
+		log.Errorf("failed to analyze spec. Spec: %s. %v", jsonSpecBytes, err)
+		return operations.NewPutAPIInventoryAPIIDSpecsProvidedSpecBadRequest().WithPayload("Spec validation failed")
 	}
-	if err := validateJSONSpec(jsonSpec); err != nil {
-		log.Errorf("Spec validation failed. Spec: %s. %v", jsonSpec, err)
+
+	// Validates an OpenAPI 2.0 specification document.
+	err = validate.Spec(analyzed, strfmt.Default)
+	if err != nil {
+		log.Errorf("spec validation failed. %v", err)
```

Comment:
You have a function for that (validateRawJSONSpec)
we have the same function in the speculator so if you really want to go all in..

Reply:
I just deleted the function because it was not needed any more here (I needed to use the `analyzed`)

Reply:
oh I see....